File: lib/detectors/googleApiKey.js
module.exports = {
  name: 'Google API Keys',
  severity: 'high',
  run(content) {
    const re = /AIza[0-9A-Za-z\-_]{35}/g;
    const seen = new Set();
    const out = [];
    let m;
    while ((m = re.exec(content)) !== null) {
      const val = m[0];
      if (seen.has(val)) continue;
      seen.add(val);
      const idx = m.index;
      const context = content.substr(Math.max(0, idx - 40), Math.min(120, val.length + 80)).replace(/\n/g, ' ');
      out.push({ value: val, context, severity: 'high' });
    }
    return out;
  }
};

File: JSX/detectors/googleApiKey.py
"""Google API key detector."""

import re
from .base import BaseDetector


class Detector(BaseDetector):
    """Detects Google API keys in JavaScript."""

    def __init__(self):
        super().__init__("Google API Keys", severity="high")
        self.pattern = re.compile(r'AIza[0-9A-Za-z\-_]{35}')

    def run(self, content):
        """Find all Google API keys."""
        findings = []
        seen = set()

        for match in self.pattern.finditer(content):
            value = match.group(0)
            if value in seen:
                continue
            seen.add(value)

            context = self._get_context(content, match.start())
            findings.append({
                "value": value,
                "context": context,
                "pos": match.start(),
                "severity": "high"
            })

        return findings
